### PR TITLE
Enable the ESLint `no-var` rule in the `src/core/evaluator.js` file

### PR DESCRIPTION
**Pull request by Snuffleupagus**
This was previously attempted in PR #13371, but had to be reverted because of issues related to SystemJS (which has since been removed).

Also, while unrelated, shortens an existing conditional assignment.

### CODE REVIEW

File: src/core/evaluator.js [hunk 1]
```diff
@@ -3239,12 +3238,10 @@ class PartialEvaluator {
...
             flushTextContentItem();
-            if (!xobjs) {
-              xobjs = resources.get("XObject") || Dict.empty;
-            }
+            xobjs ??= resources.get("XObject") || Dict.empty;
 
-            var isValidName = args[0] instanceof Name;
-            var name = args[0].name;
+            isValidName = args[0] instanceof Name;
+            name = args[0].name;
```
Comment (nicolo-ribaudo) [inline src/core/evaluator.js:3244]: Is it intentional that these two variables are now declared outside of the `promiseBody` function?

**Review by timvandermeij — APPROVED** [review]

**Comment by timvandermeij** [comment]
Good that we can do this now; thanks!

**Pull request merged**